app/models/user.rb

```diff
@@ -58,6 +58,25 @@ class User < ActiveRecord::Base
   scope :last_signed_in_after, lambda { |date| web_users.not_suspended.where('date(current_sign_in_at) >= date(?)', date) }
   scope :not_recently_unsuspended, lambda { where(['unsuspended_at IS NULL OR unsuspended_at < ?', UNSUSPENSION_GRACE_PERIOD.ago]) }
 
+  scope :with_status, lambda { |status|
+    case status
```

Comment:
@fofr just made an excellent suggestion to write a unit test that loops through `User::USER_STATUSES` and makes sure that all statuses are covered